hazelcast/src/main/java/com/hazelcast/internal/util/SortingUtil.java

```diff
@@ -189,6 +193,21 @@ public static boolean compareAnchor(PagingPredicate pagingPredicate, QueryableEn
         return SortingUtil.compare(comparator, iterationType, anchor, queryEntry) < 0;
     }
 
+    public static List<Map.Entry<Data, Data>> getSortedSubList(ArrayList<LazyMapEntry> accumulatedList,
```

Comment:
I changed so that I was able to remove the intermediate sublist. I added a method `getPageIndexesAndUpdateAnchor` which will only return the indexes to the sorted list and I directly compose the entry data list from these indexes.